### PR TITLE
Terraform: Create S3 bucket for doc upload

### PR DESCRIPTION
**Pull request by SarahJaine**
**Description of change**
- Update terraform code to create s3 buckets to use for document upload (a field in the activity report)
- Create service binding for s3 buckets

**How to test**
- Tested against sandbox. 
  - There is a new s3 bucket `ttahub-document-upload-sandbox` in that space > https://dashboard.fr.cloud.gov/services
  - And there is a new service binding for that bucket > https://dashboard.fr.cloud.gov/applications/2oBn9LBurIXUNpfmtZCQTCHnxUM/08d4c228-67c6-47e1-ac9a-bb030c880160/services

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/149

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated - documentation on service binding added in https://github.com/adhocteam/Head-Start-TTADP/pull/71
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: terraform/dev/main.tf [hunk 1]
```diff
@@ -52,3 +52,17 @@ resource "cloudfoundry_service_instance" "database" {
   space        = data.cloudfoundry_space.space.id
   service_plan = data.cloudfoundry_service.rds.service_plans["micro-psql"]
 }
+
```
Comment (jasalisbury) [inline terraform/dev/main.tf:55]: I probably say this with most terraform I review but it seems like there should be _some_ way of reducing the duplication across all our environments.

**Review by jasalisbury — APPROVED** [review]
Looks good! Only have one comment but no need to hold up this PR for it.

File: terraform/dev/main.tf (hunk 1 above)
Comment (SarahJaine) [reply terraform/dev/main.tf:55]: I can create modules, but then you have to call the module in every env and the only two lines would be deduplicated, one line if we need to change the service plan (bucket type) for any env.  Remember, this is configuration, not code.

Comment (SarahJaine) [reply terraform/dev/main.tf:55]: Also, given that this team is small and we will not be tinkering with these files ofter, I'm prioritizing readability over cleverness.

**Pull request merged**